File: src/backend/schemas/forecast.py
from decimal import Decimal

from pydantic import BaseModel


class ForecastHorizonSchema(BaseModel):
    months: int
    projected_net_funds: Decimal
    delta: Decimal


class ForecastResponse(BaseModel):
    cash_total: Decimal
    loans_total: Decimal       # negative — signed sum of loan + credit balances
    net_funds: Decimal
    avg_monthly_change: Decimal
    months_of_data: int
    data_warning: bool
    horizons: list[ForecastHorizonSchema]  # empty when months_of_data == 0

File: src/frontend/src/lib/api/forecast.ts
export interface ForecastHorizon {
  months: number;
  projected_net_funds: string;  // Decimal as string; may be negative
  delta: string;                // signed — negative when declining
}

export interface ForecastData {
  cash_total: string;
  loans_total: string;          // negative string
  net_funds: string;
  avg_monthly_change: string;
  months_of_data: number;
  data_warning: boolean;
  horizons: ForecastHorizon[];  // empty array when months_of_data === 0
}

export async function getForecast(): Promise<ForecastData> {
  const res = await fetch('/api/forecast');
  if (!res.ok) throw new Error('Failed to load forecast');
  return res.json();
}
